state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1776,7 +1872,57 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		let deposit_channel_details = DepositChannelLookup::<T, I>::get(&deposit_address)
 			.ok_or(Error::<T, I>::InvalidDepositAddress)?;
 
+		ensure!(
+			deposit_channel_details.deposit_channel.asset == asset,
+			Error::<T, I>::AssetMismatch
+		);
+
 		let channel_id = deposit_channel_details.deposit_channel.channel_id;
+		let channel_owner = deposit_channel_details.clone().owner;
+
+		if let Some(tainted_tx) =
+			TaintedTransactions::<T, I>::take(channel_owner.clone(), deposit_details.clone())
```

Comment:
nit: I think we should consider moving this check below the minimum deposit check, simply because the min deposit check is cheaper. Also, if the deposit is below the minimum, we drop it anyway (ie. we don't need to refund it) .